common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java

```diff
@@ -48,4 +48,74 @@
+     * specified expression, matching concept name, synonym, ID, or code.
+     * @param matchExp SQL MATCH expression to match concept name or synonym
+     * @param query query expression to match concept code (exactly)
+     * @param conceptId ID representation of the query (or null) for matching concept IDs
+     * @param domainId domain ID to use when filtering concepts
+     * @return per-vocabulary concept counts
+     */
+    @Query(value = "select c.vocabularyId as vocabularyId, count(distinct c.conceptId) as conceptCount from Concept c\n" +
+        "left join c.synonyms cs\n" +
+        "where (c.countValue > 0 or c.sourceCountValue > 0) and\n" +
+        "(match(c.conceptName, ?1) > 0 or\n" +
+        "match(cs.conceptSynonymName, ?1) > 0 or\n" +
+        "c.conceptId=?3 or c.conceptCode=?2) and\n" +
+        "c.standardConcept IN ('S', 'C') and\n" +
+        "c.domainId = ?4\n" +
+        "group by c.vocabularyId\n" +
+        "order by c.vocabularyId\n")
+    List<VocabularyCount> findVocabularyStandardConceptCounts(String matchExp, String query,
+        Long conceptId, String domainId);
+
+    /**
+     * Return the number of concepts (standard or non-standard) in each vocabulary
+     * for the specified domain matching the specified expression, matching concept name, synonym,
+     * ID, or code.
+     * @param matchExp SQL MATCH expression to match concept name or synonym
+     * @param query query expression to match concept code (exactly)
+     * @param conceptId ID representation of the query (or null) for matching concept IDs
+     * @param domainId domain ID to use when filtering concepts
+     * @return per-vocabulary concept counts
+     */
+    @Query(value = "select c.vocabularyId as vocabularyId, count(distinct c.conceptId) as conceptCount from Concept c\n" +
+        "left join c.synonyms cs\n" +
+        "where (c.countValue > 0 or c.sourceCountValue > 0) and\n" +
+        "(match(c.conceptName, ?1) > 0 or\n" +
+        "match(cs.conceptSynonymName, ?1) > 0 or\n" +
+        "c.conceptId=?3 or c.conceptCode=?2) and\n" +
+        "c.domainId = ?4\n" +
+        "group by c.vocabularyId\n" +
+        "order by c.vocabularyId\n")
+    List<VocabularyCount> findVocabularyAllConceptCounts(String matchExp, String query, Long conceptId, String domainId);
+
+    /**
+     * Return the number of standard concepts in each vocabulary for the specified domain.
+     * @param domainId domain ID to use when filtering concepts
+     * @return per-vocabulary concept counts
+     */
+    @Query(value = "select c.vocabularyId as vocabularyId, count(distinct c.conceptId) as conceptCount from Concept c\n" +
+        "where (c.countValue > 0 or c.sourceCountValue > 0) and\n" +
+             "c.standardConcept IN ('S', 'C') and\n" +
+        "c.domainId = ?1\n" +
+        "group by c.vocabularyId\n" +
+        "order by c.vocabularyId\n")
+    List<VocabularyCount> findVocabularyStandardConceptCountsInDomain(String domainId);
```

Comment:
Note: the results to these next two methods can and should probably be cached in the database, rather than calculated on the fly. I'll look into doing that in a follow up PR, at which point I'll get rid of these methods.

Reply:
While I see a little bit of value in not having to make schema changes every time we want some new rollup, I personally don't love throwing everything in achilles_results, for a few reasons:

1) It is a hodge podge of different counts for different things... it's very hard to document and understand how to use it in code.
2) JPA doesn't work very well with it, particularly WRT child tables. I can't for example have a List<DomainVocabularyInfo> that gets populated when I load a DomainInfo based on AchillesResult rows without writing a bunch of JQL.
3) Performance-wise, pulling from a small table that has just what you want will always be faster than joining one or more times into a larger table.

So I'd prefer we go with the domain_vocabulary_info table in the PR I just merged. Most of the work involved is not the schema changes, but the SQL to populate it.

Reply:
So i disagree. It is not good for everything.  But it is intended to store counts based on different parameters and perfect for that. It is easy to use ... and actually pretty simple and self documenting in the achilles_analysis table.   I think that I should demo what i am talking about on a call.  

I missed that domain_vocabulary_info table merge. I was wonderding where the yaml changes were to conceptListResponse . ..